### DISCUSSION TITLE
Quality Gates, Code Analysis

### DISCUSSION THREAD

**Discussion by ggnaegi** [ThreeMammals/Ocelot#1759] [Ideas]
## Expected Behavior / New Feature

It would be great to have feedback from a static analysis tool (e.g. sonarqube) when creating new PRs.  With predefined quality gates (80% coverage on new code, no bugs, security reviewed), this would make life easier for reviewers. Reviewers wouldn't review any code until the quality gate is reached.

## Actual Behavior / Motivation for New Feature

Some PRs are pushed without unit and acceptance tests, which means a lot of work for code reviewers. Code quality is not verified either, and standards are not followed.

## Specifications

I have written a github action for sonarqube - a fantastic tool made in Switzlerand - and this could be reused here... 😸
Coverage and Code quality can be improved: https://sonarcloud.io/summary/overall?id=ggnaegi_Ocelot. - we are not far from 80%, since some files shouldn't be part of analysis -.

So, please @TomPallister, @raman-m, @RaynaldM, let's use sonarqube 😄

**Comment by raman-m** [comment]
Thanks for the feedback of your testing! 

To be fair, I do not care about quality gates now, because it requires to do a lot of tasks, and it will take much time, and benefit will be an additional step of building pipeline.
This issue is very abstract, this feature request is very abstract, and I expect a long thread of discussions...

You are enjoying new features development, but personally I have 40 open pull requests and ~450 open issues now. And nobody helps me.
If you propose to integrate SonarCube, Let me dream about it too, having broken Coveralls indicator. Also, we have nice coverage reporting lib, and it would be great to research it first.

Finally, I would prefer to recover old indicators first, make them working, after that we can plan migration or additional integration of extra-analyzers. Sounds like a good plan?

**Reply by ggnaegi** [reply]
Well, I know, but basically I'm offering a working solution that would help you and others. I've shown you that code coverage with coveralls hasn't worked for a long time now (2018!) and that probably the code available doesn't work. The addins used in cake for coveralls haven't been maintained for years now (2020).

**Reply by ggnaegi** [reply]
There is no abstraction in this, it's an indicator. For now we don't have any indicators, it's better than nothing.

**Reply by raman-m** [reply]
🆗 🆗 Let's discuss your idea after October's release
